core/src/main/java/org/apache/druid/data/input/InputEntity.java

```diff
@@ -136,4 +136,13 @@ public void close()
   {
     return Predicates.alwaysFalse();
   }
+
+  /**
+   * This is required so that an empty iterator is created for the reader corresponding to a tombstone
+   * @return false if the entity does not come from a tombstone segment, true otherwise
+   */
+  default boolean isFromTombstone()
```

Comment:
Please do not add this to the interface.  `DruidSegmentReader` and `DruidSegmentInputFormat` pretty much only know how to deal with DruidInputEntity objects anyway (it's gonna have a really hard time reading an ORC based one, for example).  As such, the users can just do an `instanceof` check and then call the concrete method directly.

Reply:
Fixed in next commit